CMakeLists.txt

```diff
@@ -1022,8 +1025,8 @@ if(MSVC)
         COMPILE_FLAGS -DYY_NO_UNISTD_H)
 endif()
 
-list(APPEND libcsound_private_include_dirs ${CMAKE_CURRENT_BINARY_DIR})
-list(APPEND libcsound_public_include_dirs $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>)
+target_include_directories(libcsound_internals INTERFACE ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Looking at this, it should use the build_interface as well if it's accessing the build folder?

Reply:
I don't think so, the `BUILD_INTERFACE` / `INSTALL_INTERFACE` bit is only necessary for public include dirs, not internals

Reply:
It doesn't make sense to include the bin directory for the configure though. It's including it when there is nothing there. If there is no build error then this must not be needed. Either way we should investigate it more.

Reply:
We're putting a couple generated files into the bin folder: https://github.com/csound/csound/blob/a1580f9cdf331c35dceb486f4231871ce0b00266/include/CMakeLists.txt#L5 https://github.com/csound/csound/blob/a1580f9cdf331c35dceb486f4231871ce0b00266/CMakeLists.txt#L972. I'm not sure that's necessarily the right place to put them though

Reply:
Ok after reading a bit more I'm not really sure now. Need to rethink which needs to be in `BUILD_INTERFACE` or not. I thought it was related to the build step but it's when exporting the build which makes sense. Lets leave it as it is and see if we encounter any problems. 

```
$<BUILD_INTERFACE:...>
Content of ... when the property is exported using [export()](https://cmake.org/cmake/help/latest/command/export.html#command:export), or when the target is used by another target in the same buildsystem. Expands to the empty string otherwise.
```